lib/data_distributor/cpr/person.rb

```diff
@@ -0,0 +1,83 @@
+module DataDistributor
+  module CPR
+    #This class represents a person as known by cpr
+    class Person
+      attr_reader :data, :cpr
+
+      # @param [Dataset]
+      def initialize(data, cpr:)
+        @data = data
+        @cpr = cpr
+      end
+
+      # Person's first name(s)
+      def first_name
+        data[:Person][:Navn][:fornavne]
+      end
+
+      # Person's last name
+      def last_name
+        data[:Person][:Navn][:efternavn]
+      end
+
+      # returns a Date object with the full date of birth, including century, which is not apparent in the cpr-number
+      def date_of_birth
+        @date_of_birth ||= begin
+          day = cpr[0..1].to_i
+          month = cpr[2..3].to_i
+          year = cpr[4..5].to_i
+          century_digit = cpr[6].to_i
+
+          case century_digit
+          when 0..3
+            year += 1900
+          when 4
+            if year <= 36
+              year += 2000
+            else
+              year += 1900
+            end
+          when 5..8
+            if year <= 57
+              year += 2000
+            else
+              year += 1800
+            end
+          when 9
+            if year <= 36
+              year += 2000
+            else
+              year += 1900
+            end
+          end
+
+          Date.new(year, month, day)
+        end
+      end
+
+      # Person's age in years
+      def age
```

Comment:
```suggestion
      # @return [Integer]
      def age_in_years
```
Det er lidt smag og behag men så er man da ikke i tvivl om enheden

Reply:
God ide